### PR TITLE
matrix: add method to check all values are nan

### PR DESCRIPTION
**Pull request by mrivi**

### CODE REVIEW

File: matrix/Matrix.hpp [hunk 1]
```diff
@@ -507,6 +507,17 @@ class Matrix
         return min_val;
     }
 
+    bool isAllNan() {
```
Comment (jkflying) [inline matrix/Matrix.hpp]:
```suggestion
    bool isAllNan() const {
```

File: matrix/Matrix.hpp [hunk 2]
```diff
@@ -507,6 +507,17 @@ class Matrix
...
+      Matrix<float, M, N> &self = *this;
```
Comment (jkflying) [inline matrix/Matrix.hpp]:
```suggestion
      const Matrix<float, M, N> &self = *this;
```

File: test/matrixAssignment.cpp [hunk 3]
```diff
@@ -30,6 +30,7 @@ int main()
     for(int i=0; i<9; i++) {
         TEST(isnan(m_nan.data()[i]));
     }
+    TEST(m_nan.isAllNan());
```
Comment (jkflying) [inline test/matrixAssignment.cpp]: Also add a test for the negative result, to make sure it doesn't always return `true`
